content/existential-containers.md

```diff
@@ -0,0 +1,206 @@
+
+The semantics of the proposed feature is fully expressible in Scala.
+Save for the implicit addition of `.value` on method selection when the receiver is an instance of `Containing[C]`, this proposal requires no change in the language.
+As a result, it has no backward binary or TASTy compatibility consequences.
+
+### Feature interactions
+
+The proposed feature is meant to interact with implicit search, as currently implemented by the language.
+More specifically, given an existential container `c`, accessing `c.value` _opens_ the existential while retaining its type `c.Value`, effectively keeping an _anchor_ (i.e., the path to the scope of the witness) to the interface of the type class.
+
+Since no change in implicit resolution is needed, this proposal cannot create unforeseen negative interactions with existing features.
+
+### Other concerns
+
+This document has been written under the experimental modularity improvements for Scala 3.
+Although the proposed feature is fully expressible without those changes, the encoding of existential containers can only work with the "old" (i.e., the one currently used in production) or "new" type class style.
+
+### Open questions
+
+One problem not addressed by the proposed encoding is the support of multiple type classes to form the interface of a specific container.
+For example, one may desire to create a container of values whose types conform to both `Polygon` _and_ `Show`.
+We have explored possible encodings for such a feature but decided to remove them from this proposal, as support for multiple type classes can most likely be achieved without any additional language change.
+
+Another open question relates to possible language support for shortening the expression of a container type and/or value.
+
+## Related work
+
+Swift supports existential containers.
+For instance, `largest` can be written as follows in Swift:
+
+```swift
+func largest(_ xs: [any Polygon]) -> (any Polygon)? {
+  xs.max { (a, b) in a.area < b.area }
+}
+```
+
+Unlike in this proposal, existential containers in Swift are built-in and have a dedicated syntax (i.e., `any P`).
+One advantage of Swift's design is that the type system can treat an existential container as supertype of types conforming to that container's interface.
+For example, `any Polygon` is supertype of `Square` (assuming the latter conforms to `Polygon`):
+
+```swift
+print(largest([Square(), Hexagon()]))
+```
+
+In contrast, to avoid possible undesirable complications, this proposal does not suggest any change to the subtyping relation of Scala.
+
+Rust also supports existential containers in a similar way, writing `dyn P` to denote a container bundling some value of a type conforming to `P`.
+Similar to Swift, existential containers in Rust are considered supertypes of the types conforming to their bound.
+
+
+A more formal exploration of the state of the art as been documented in a research paper presented prior to this SIP [2].
+
```

Comment:
This proposal is missing a section on `Alternatives`. I can think of a few myself:

* What if we just add a `Containing[T]` type to the standard library; people could then use it without needing a language/compiler change at all?

* What about implicit conversions and implicit constructors? Those are widely used today, often referred to as the "magnet pattern"